src/typings/prop-types.d.ts

```diff
@@ -0,0 +1,7 @@
+declare module 'prop-types' {
+    import React = require('react');
+
+    export const bool: {
+        isRequired: React.Validator<any>
```

Comment:
FYI, Limiting the typings doens't actually limit the scope of what's imported, it just limits what the TS compiler will let you use.
I suspect if you remove all references to React.PropTypes internally in react.d.ts the warning will go away.
Ideally we would be consuming all our typings files directly from @types repo